ui/com/user-info.jsx

```diff
@@ -25,8 +25,8 @@ const FLAG_DROPDOWN = [
 class AutoRefreshingComponent extends React.Component {
   constructor(props) {
     super(props)
-    this.state = this.computeState()
-    this.refreshState = props => { this.setState(this.computeState(props)) }
+    this.computeState(null, this.setStateCallback.bind(this))
```

Comment:
Oh, no you can declare a constant function inside the constructor. Let me illustrate the pattern via es5:

```js
function foo() {
  function bar() {
    // stuff
  }
  bar()
}
```

So that's the basic pattern, a function within a function. And then, even in es5, you could do it as a variable assignment, like so:

```js
function foo() {
  var bar = function () {
    // stuff
  }
  bar()
}
```

So now, with es6, I use `const` instead of `var` whenever possible, to show when I do/dont expect a variable to mutate:

```js
function foo() {
  const bar = function () {
    // stuff
  }
  bar()
}
```

Then I use the fat-arrow syntax, partially for readability, but also for the `this` context:

```js
function foo() {
  const bar = () => {
    // stuff
  }
  bar()
}
```

The fat-arrow declaration changes the closure rules for javascript, so that it takes the `this` of the closure it was declared in. Without it, we'd have to do:

```js
function foo() {
  const bar = (function() {
    // stuff
  }).bind(this)
  bar()
}
```

or something similar. And then, lastly, we have the es6 class stuff, which is why we have a `this` to care about at all.

So here's what I was suggesting:

```js
class AutoRefreshingComponent extends React.Component {
  constructor(props) {
    super(props)
    const handleCB = (err, state) => {
      if (err) throw err
      this.setState(state)
    }
    this.computeState(null, handleCB)
    this.refreshState = props => { this.computeState(props, handleCB) }
  }
...
```

Probably more explanation than you needed, but I figured I'd err on the side of clarity :)